Expensify/Bedrock
# Add docker support

This pull request install the Docker configuration for Bedrock project, as well as the build and extract script.

## Changes made

### docker-compose.yml
- Added configuration for a 3-node Bedrock cluster
- Set up networks and volumes for each node
- Defined node-specific environment variables

### Dockerfile
- Implemented a two-stage build to optimize final image size
- Used Ubuntu 22.04 as the base for the build stage
- Installed and configured apt-fast to speed up downloads
- Installed Clang 18 and necessary dependencies
- Configured ccache to improve compilation times
- Compiled the Bedrock project and ran tests
- Used phusion/baseimage:noble-1.0.0 for the final image
- Configured the Bedrock service to run with proper permissions

### build_and_extract.sh
- Added a script to build the Docker image and extract the Bedrock binary
- The script creates a temporary container, copies the binary, and places it in the output directory

## Motivation
These changes aim to improve our Docker build process, facilitate Bedrock cluster deployment, and optimize the size and performance of our Docker images.

## How to test
1. Ensure you have Docker and Docker Compose installed on your machine
2. Clone this branch and cd into `docker` folder
3. Run `docker-compose up --build` to build and start the Bedrock cluster
4. Verify that all three nodes start correctly and communicate with each other
5. Run the `build_and_extract.sh` script and check that the Bedrock binary is correctly extracted

Please feel free to provide any feedback or suggestions for improvement!
